Realm/Tests/TransactionTests.m

```diff
@@ -52,4 +83,102 @@ - (void)testTransactionMisuse
     XCTAssertThrows([realm deleteObject:obj], @"Outside writetransaction");
 }
 
+- (void)testDeleteBug
+{
+    RLMRealm *realm = [RLMRealm defaultRealm];
+    [realm transactionWithBlock:^{
+        [realm deleteAllObjects];
+    }];
+    
+    Breeder *breeder = [[Breeder alloc] init];
+    [realm transactionWithBlock:^{
+        [realm addObject:breeder];
+    }];
+    
+    RLMResults *results = [Dog objectsInRealm:realm where:@"ANY breeders = %@", breeder];
+    
+    [realm transactionWithBlock:^{
+        Person *person = [[Person alloc] init];
+        
+        Dog *dog = [[Dog alloc] init];
+        person.dog = dog;
+        
+        [realm addObject:person];
+    }];
+    
+    [realm transactionWithBlock:^{
+        Breeder *breeder = [Breeder allObjectsInRealm:realm].firstObject;
+        if (breeder) {
+            [realm deleteObject:breeder];
+        }
+    }];
+    XCTAssertNoThrow(results.count, @"Delete test Failed.");
+}
+
+- (void)testMovedItems
+{
+    RLMRealm *realm = [RLMRealm defaultRealm];
+    [realm transactionWithBlock:^{
+        [realm deleteAllObjects];
+    }];
+    
+    Breeder *breeder = [[Breeder alloc] init];
+    [realm transactionWithBlock:^{
+        [realm addObject:breeder];
+    }];
+    
+    Breeder *second_breeder = [[Breeder alloc] init];
+    [realm transactionWithBlock:^{
+        [realm addObject:second_breeder];
+    }];
+    
+    RLMResults *results = [Dog objectsInRealm:realm where:@"ANY breeders = %@", breeder];
+
+    [realm transactionWithBlock:^{
+        Dog *dog = [[Dog alloc] init];
+        [dog.breeders addObject:breeder];
+        [realm addObject:dog];
+        
+        Dog *second_dog = [[Dog alloc] init];
+        [second_dog.breeders addObject:second_breeder];
+        [realm addObject:second_dog];
+        
+        [realm deleteObject:breeder];
+        
+    }];
+    
+    XCTAssertNoThrow(results.count, @"Moved Items test Failed.");
+    XCTAssertEqual(results.count,(unsigned long)0);
+}
+
+- (void)testLinkQueryWhenLinkIsDeleted
```

Comment:
My reading is that this single test exercises the same code paths as the two preceding tests. Do the other two tests exercise any behavior that this one doesn't? If not, it seems like we'd only need the one test case, and can avoid adding the new model types.